linkerd/app/outbound/src/endpoint.rs

```diff
@@ -68,64 +67,52 @@ impl<E> Outbound<E> {
     }
 
     /// Pushes a layer that checks if a discovered service profile contains an
-    /// endpoint override, and forwards directly to that endpoint (bypassing the
-    /// current stack) if one exists.
-    pub fn push_endpoint_override<T, O, R>(
+    /// endpoint, and forwards directly to that endpoint (bypassing the current
+    /// stack) if one exists.
+    pub fn push_profile_endpoint<T, E, ESvc, NSvc, Req>(
         self,
-        ep_override: O,
+        endpoint: E,
     ) -> Outbound<
         impl svc::NewService<
                 (Option<profiles::Receiver>, T),
-                Service = OrOverride<E::Service, O::Service>,
+                Service = svc::stack::ResultService<svc::Either<NSvc, ESvc>>,
```

Comment:
unrelated thought: might be nice to export a `Switch` type alias for `ResultService<Either<A, B>>` in `svc`, which is a bit more concise than writing out the whole type but doesn't require each stack layer to define a custom alias...not a big deal.